digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/linearasset/LinearAssetFiller.scala

```diff
@@ -1,14 +1,56 @@
 package fi.liikennevirasto.digiroad2.linearasset
 
-import fi.liikennevirasto.digiroad2.asset.{SideCode}
+import fi.liikennevirasto.digiroad2.asset.{DynamicProperty, SideCode}
 
 object LinearAssetFiller {
   case class MValueAdjustment(assetId: Long, linkId: Long, startMeasure: Double, endMeasure: Double)
   case class VVHChangesAdjustment(assetId: Long, linkId: Long, startMeasure: Double, endMeasure: Double, vvhTimestamp: Long)
   case class SideCodeAdjustment(assetId: Long, sideCode: SideCode, typeId: Int)
+  case class ValueAdjustment(asset: PersistedLinearAsset)
   case class ChangeSet(droppedAssetIds: Set[Long],
                        adjustedMValues: Seq[MValueAdjustment],
                        adjustedVVHChanges: Seq[VVHChangesAdjustment],
                        adjustedSideCodes: Seq[SideCodeAdjustment],
-                       expiredAssetIds: Set[Long])
+                       expiredAssetIds: Set[Long],
+                       valueAdjustments: Seq[ValueAdjustment])
 }
+
+//package fi.liikennevirasto.digiroad2.linearasset
+//
+//import fi.liikennevirasto.digiroad2.asset.SideCode
+//import org.joda.time.DateTime
+//
+////object LinearAssetFiller {
+////  case class MValueAdjustment(assetId: Long, linkId: Long, startMeasure: Double, endMeasure: Double)
+////  case class VVHChangesAdjustment(assetId: Long, linkId: Long, startMeasure: Double, endMeasure: Double, vvhTimestamp: Long)
+////  case class SideCodeAdjustment(assetId: Long, sideCode: SideCode, typeId: Int)
+////  case class ChangeSet(droppedAssetIds: Set[Long],
+////                       adjustedMValues: Seq[MValueAdjustment],
+////                       adjustedVVHChanges: Seq[VVHChangesAdjustment],
+////                       adjustedSideCodes: Seq[SideCodeAdjustment],
+////                       expiredAssetIds: Set[Long])
+////}
+//
+//trait Filler {
+//  case class MValueAdjustment(assetId: Long, linkId: Long, startMeasure: Double, endMeasure: Double)
+//  case class VVHChangesAdjustment(assetId: Long, linkId: Long, startMeasure: Double, endMeasure: Double, vvhTimestamp: Long)
+//  case class SideCodeAdjustment(assetId: Long, sideCode: SideCode, typeId: Int)
+//
+//}
+//
+//object LinearAssetFiller extends Filler {
+//  case class ChangeSet(droppedAssetIds: Set[Long],
+//                       adjustedMValues: Seq[MValueAdjustment],
+//                       adjustedVVHChanges: Seq[VVHChangesAdjustment],
+//                       adjustedSideCodes: Seq[SideCodeAdjustment],
+//                       expiredAssetIds: Set[Long])
+//}
+//
+//object MyFiller extends Filler{
+//  case class ChangeSet(droppedAssetIds: Set[Long],
+//                       adjustedMValues: Seq[MValueAdjustment],
+//                       adjustedVVHChanges: Seq[VVHChangesAdjustment],
+//                       adjustedSideCodes: Seq[SideCodeAdjustment],
+//                       expiredAssetIds: Set[Long],
+//                       adjustedDate: Set[DateTime])
+//}
```

Comment:
Commented code

Reply:
I removed the commented code, it was originally a thought to refactor the LinearAssetFiller object to support multiple ChangeSets. I wanted to add the valueAdjustments-parameter, but not for all ChangeSets, just for the ones that need it. If you have any ideas on how to do it, please tell me. Otherwise it can just be passed as empty